api/src/pcapi/core/users/api.py

```diff
@@ -431,22 +436,54 @@ def suspend_account(
             user.backoffice_profile.roles = []
 
     n_bookings = 0
+    if backoffice:
+        cancel_bookings = False
+        filters = []
+        if reason in (constants.SuspensionReason.FRAUD_RESELL_PRODUCT, constants.SuspensionReason.FRAUD_RESELL_PASS):
+            cancel_bookings = True
+        if reason in (
+            constants.SuspensionReason.FRAUD_HACK,
+            constants.SuspensionReason.SUSPICIOUS_LOGIN_REPORTED_BY_USER,
+            constants.SuspensionReason.UPON_USER_REQUEST,
+        ):
+            cancel_bookings = True
+            filters.append(sa.func.not_(offers_models.Offer.isEvent))
+
+        if cancel_bookings:
+            bookings = (
+                bookings_models.Booking.query.join(bookings_models.Booking.stock)
+                .join(offers_models.Stock.offer)
+                .filter(
+                    bookings_models.Booking.userId == user.id,
+                    bookings_models.Booking.status == bookings_models.BookingStatus.CONFIRMED,
+                    sa.or_(
+                        datetime.datetime.utcnow() < bookings_models.Booking.cancellationLimitDate,
+                        bookings_models.Booking.cancellationLimitDate.is_(None),
+                    ),
```

Comment:
Souhaite-t-on effectivement ne pas annuler si `cancellationLimitDate` est passée ? C'est la date limite d'annulation par l'utilisateur, mais pas forcément pour fraude.
De plus, si on garde cette condition, je ne la mettrais que dans le cas `if reason in (constants.SuspensionReason.FRAUD_RESELL_PRODUCT, constants.SuspensionReason.FRAUD_RESELL_PASS)` car on ne prend pas les évènements dans l'autre cas.

Reply:
j'ai repris le fonctionnement de `_cancel_bookings_of_user_on_requested_account_suspension` a ce niveau là comme ce n'était pas spécifié dans le ticket j'ai gardé le comportement.